benchmarks/winogrande/winogrande.py

```diff
@@ -0,0 +1,124 @@
+"""WINOGRANDE: An Adversarial Winograd Schema Challenge at Scale
+
+Keisuke Sakaguchi, Ronan Le Bras, Chandra Bhagavatula, Yejin Choi
+https://arxiv.org/pdf/1907.10641
+
+The prompt template is based on the multiple choice template in openai simple evals:
+https://github.com/openai/simple-evals/blob/main/mmlu_eval.py
+
+# run with default fewshot samples (5)
+inspect eval winogrande/winogrande.py
+
+# run with more or no fewshot samples
+inspect eval winogrande/winogrande.py -T fewshot=5
+inspect eval winogrande/winogrande.py -T fewshot=false
+
+# run with different subset of the dataset
+inspect eval winogrande/winogrande.py -T dataset_name=winogrande_l
+"""
+
+from typing import Dict
+
+from inspect_ai import Task, task
+from inspect_ai.dataset import Sample, hf_dataset
+from inspect_ai.model import GenerateConfig
+from inspect_ai.scorer import choice
+from inspect_ai.solver import (
+    multiple_choice,
+    system_message,
+)
+
+# Assumption: "A" and "B" are the option letters.
+ANSWER_TO_LETTER = {"1": "A", "2": "B"}
+BLANK_TAG = "[BLANK]"
+
+WINOGRANDE_PROMPT_TEMPLATE = {
+    # Based on the prompt provided here:
+    # https://huggingface.co/datasets/meta-llama/Meta-Llama-3.1-8B-evals/viewer/Meta-Llama-3.1-8B-evals__winogrande__details?row=0
+    "system_w_examples": f"""The following are multiple choice questions, with answers on the best logical completion to replace {BLANK_TAG} by {list(ANSWER_TO_LETTER.values())[0]} or {list(ANSWER_TO_LETTER.values())[1]}."""
+    + """\n\n{examples}\n\n""",
+    # Based on the SINGLE_ANSWER_TEMPLATE provided in the multiple choice solver:
+    # https://github.com/UKGovernmentBEIS/inspect_ai/blob/main/src/inspect_ai/solver/_multiple_choice.py#L14
+    "user": f"""Answer the following multiple choice question by choosing the best logical option to replace the {BLANK_TAG}."""
+    + """ The entire content of your response should be of the following format: 'ANSWER: $LETTER' (without quotes) where LETTER is one of {letters}.\n\n{question}\n{choices}\n""",
+}
+
+
+@task
+def winogrande(
+    dataset_name: str = "winogrande_xl",
+    fewshot: int = 5,
+    fewshot_seed: int = 42,
+) -> Task:
+    """Inspect task implementing the WinoGrande benchmark.
+
+    Arguments:
+        dataset_name (str): Subset of the dataset to be used.
+        fewshot (int): Number of few shot examples to use.
+        fewshot_seed (int): Random seed for sampling few shot examples.
+    """
+    # Get prompt
+    prompt_dict = WINOGRANDE_PROMPT_TEMPLATE
+    prompt_user = prompt_dict["user"]
+    prompt_sys_w_examples = prompt_dict["system_w_examples"]
+
+    plan = [multiple_choice(template=prompt_user, shuffle=False)]
+    scorer = choice()
+    gen_config = GenerateConfig(max_tokens=64)
+
+    if fewshot:
```

Comment:
Maybe here we create a function that yields the `plan` (and takes the few shot args). I think seeing all of the plan building code together will be a plus and clearing out the few shot code from the main flow will make things more readable)

Reply:
Created a function for building the plan, similar to the DROP task.